### PR TITLE
feat: add version function to the client.

### PR DESCRIPTION
**Pull request by ghyatzo**
can be useful to have, not sure if it is the correct place to put it though.
I thought that going `codemp::Client::version()` was good enough.

### CODE REVIEW

**Comment by zaaarf** [comment]
1. I think it should be moved as a function under lib.rs rather than as a client method. 
2. I don't think we should bother with the crate name and license. Those are unlikely to change; the string the user will be interested in is just the string.
3. Before we merge this, I think we should all add the respective glue code to expose it. I can already say that the Java glue will have to expose this either under `Client.java` or `Extensions.java`, regardless of what we decide on point 1, because everything must belong to a class there.

**Comment by alemidev** [comment]
This is a wise addition but should probably be more machine friendly in its format: while most uses will just be a display flag to know which version you're running, I think this function should provide means to do crude version checking downstream, maybe to ensure the correct shared library version has been loaded. A "vX.Y.Z" string may be better, or maybe just a "X.Y.Z". A numeric data structure would probably be better (@zaaarf mentioned tuple of 3x u32) but is likely a lot less convenient to pass across FFI

File: src/client.rs [hunk 1]
```diff
@@ -165,4 +165,13 @@ impl Client {
...
 		&self.0.user
 	}
+
+	/// Get the current version of the client
+	pub fn version() -> String {
+		let version = env!("CARGO_PKG_VERSION");
+		let name = env!("CARGO_PKG_NAME");
+		let license = env!("CARGO_PKG_LICENSE");
+
+		format!("{name} - {version} ({license})")
```
Comment (alemidev) [inline src/client.rs]: this should probably just return a flat `env!("CARGO_PKG_VERSION")`, our package name and license are stable and can be formatted in on the receiving end

**Comment by ghyatzo** [comment]
In my opinion, for now keeping it simple with just a semver string would be best.
In the future if the need for extensive version checking arises we can expand to a structure.

**Review by alemidev — DISMISSED** [review]

**Review by alemidev — DISMISSED** [review]

**Review by alemidev — APPROVED** [review]

**Pull request merged**